### PR TITLE
Allows copyImplementation() to work with a dest top cell.

### PR DESCRIPTION
**Pull request by clavin-xlnx**
This was originally included in #385, but it has been isolated here as its own PR.

### CODE REVIEW

File: src/com/xilinx/rapidwright/design/DesignTools.java [hunk 1]
```diff
@@ -2152,6 +2153,17 @@ public static void copyImplementation(Design src, Design dest, boolean lockPlace
 		HashSet<String> instsWithSeparator = new HashSet<>();
 		for(Entry<String,String> e : srcToDestInstNames.entrySet()) {
 			EDIFHierCellInst cellInst = src.getNetlist().getHierCellInstFromName(e.getKey());
+			if(e.getValue().length() == 0) {
+			    // If its the top cell, remove the top cell from destNetlist
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/design/DesignTools.java:2157]: Add a line to the javadoc stating that an empty destination string indicates copying to the top cell?

Comment (eddieh-xlnx) [reply src/com/xilinx/rapidwright/design/DesignTools.java:2157]: Perhaps the doc should also mention that destination cell instances that are already present will be replaced?

File: src/com/xilinx/rapidwright/design/DesignTools.java [hunk 2]
```diff
@@ -2238,6 +2248,13 @@ public static void copyImplementation(Design src, Design dest, boolean lockPlace
 		}		
 	}
 	
+	private static String getNewHierName(String srcName, Map<String,String> srcToDestInstNames, 
+	                                        Map<String,String> prefixes, String prefixMatch) {
+        String newCellPrefix = srcToDestInstNames.get(prefixes.get(prefixMatch));
+        int idx = prefixMatch.length() - (newCellPrefix.length() == 0 ? 0 : 1);
+        return newCellPrefix + srcName.substring(idx);
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/design/DesignTools.java:2255]:
```suggestion
		String newCellPrefix = srcToDestInstNames.get(prefixes.get(prefixMatch));
		int idx = prefixMatch.length() - (newCellPrefix.length() == 0 ? 0 : 1);
		return newCellPrefix + srcName.substring(idx);
```

**Review by eddieh-xlnx — APPROVED** [review]

File: src/com/xilinx/rapidwright/design/DesignTools.java (hunk 1 above)
Comment (clavin-xlnx) [reply src/com/xilinx/rapidwright/design/DesignTools.java:2159]: Updated.

**Review by eddieh-xlnx — APPROVED** [review]

File: src/com/xilinx/rapidwright/design/DesignTools.java [hunk 3]
```diff
@@ -2129,9 +2130,11 @@ public static Map<String,String> getInvertiblePinMap(Series series, Unisim unisi
 	}
 
 	/**
-	 * Copies the logic and implementation of a set of cells from one design to another.
+	 * Copies the logic and implementation of a set of cells from one design to another.  This will
+	 * replace the destination logical cell instances with those of the source design. 
 	 * @param src The source design (with partial or full implementation)
-	 * @param dest The destination design (with matching cell instance interfaces) 
+	 * @param dest The destination design (with matching cell instance interfaces). If targeting
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/design/DesignTools.java:2136]: Wait. Are you sure it's this param that is expected to be empty (and not the `srcToDestInstNames` param)

Comment (clavin-xlnx) [reply src/com/xilinx/rapidwright/design/DesignTools.java]: Sure, moved it to the other param.

**Review by eddieh-xlnx — APPROVED** [review]

**Pull request merged**